### PR TITLE
fix: do not call logout on delete member as the backend does it automatically

### PR DESCRIPTION
**Pull request by spaenleh**
Linked to https://github.com/graasp/graasp/pull/1158

### CODE REVIEW

File: src/member/api.ts [hunk 1]
```diff
@@ -112,6 +113,16 @@ export const deleteMember = async (
...
   );
 
+export const deleteCurrentMember = async ({
+  API_HOST,
+  axios,
+}: PartialQueryConfigForApi) =>
+  verifyAuthentication(() =>
+    axios
+      .delete<void>(`${API_HOST}/${buildDeleteCurrentMemberRoute()}`)
+      .then(({ data }) => data),
```
Comment (pyphilia) [inline src/member/api.ts:123]: If void do you need to get the data?

File: src/member/mutations.ts [hunk 2]
```diff
@@ -54,6 +53,35 @@ export default (queryConfig: QueryClientConfig) => {
...
+        });
+
+        queryClient.resetQueries();
+
+        // remove cookies from browser when logout succeeds
+        if (queryConfig.DOMAIN) {
+          // todo: find a way to do this with an httpOnly cookie
+          // removeSession(id, queryConfig.DOMAIN);
+          // setCurrentSession(null, queryConfig.DOMAIN);
+        }
```
Comment (pyphilia) [inline src/member/mutations.ts:72]: remove?

File: src/member/mutations.ts [hunk 3]
```diff
@@ -54,6 +53,35 @@ export default (queryConfig: QueryClientConfig) => {
...
+
+        // Update when the server confirmed the logout, instead optimistically updating the member
+        // This prevents logout loop (redirect to logout -> still cookie -> logs back in)
+        queryClient.setQueryData(memberKeys.current().content, undefined);
+      },
+      // If the mutation fails, use the context returned from onMutate to roll back
+      onError: (error: Error, _args, _context) => {
+        notifier?.({ type: deleteMemberRoutine.FAILURE, payload: { error } });
```
Comment (pyphilia) [inline src/member/mutations.ts:80]:
```suggestion
        notifier?.({ type: deleteCurrentMemberRoutine.FAILURE, payload: { error } });
```

**Review by pyphilia — APPROVED** [review]
Thanks for the PR, some changes would be welcome, otherwise it's good to go! 📗

File: src/member/api.ts (hunk 1 above)
Comment (spaenleh) [reply src/member/api.ts:123]: It is needed, so the return type is `Promise<void>` and not `Promise<AxiosResponse<void>>`

**Pull request merged**